proto/tendermint/abci/types.proto

```diff
@@ -258,6 +297,32 @@ message ResponseApplySnapshotChunk {
   }
 }
 
+message ResponseExtendVote {
+  tendermint.types.VoteExtension vote_extension = 1;
+}
+
+message ResponseVerifyVoteExtension {
+  Result result = 1;
+
+  enum Result {
+    UNKNOWN = 0;  // Unknown result, treat as ACCEPT by default
+    ACCEPT  = 1;  // Vote extension verified, include the vote
+    SLASH   = 2;  // Vote extension verification aborted, continue but slash validator
+    REJECT  = 3;  // Vote extension invalidated
+  }
+}
+
+message ResponsePrepareProposal {
+  repeated bytes block_data = 1;
```

Comment:
don't we also include the max block size in here?

Reply:
Tendermint already knows the max size. we double check the response size when its returned.

Reply:
https://github.com/tendermint/tendermint/blob/f408849112ab06ecafd6f4a41ed908e3f5b6907c/proto/tendermint/abci/types.proto#L133-L140

Reply:
we send the size to the app. We dont need it back. What is the question?

Reply:
lol my bad. Got the request and response mixed up. Ignore this